src/Symfony/Component/Config/Definition/Dumper/YamlReferenceDumper.php

```diff
@@ -89,8 +89,8 @@ private function writeNode(NodeInterface $node, ?NodeInterface $parentNode = nul
                 $children = $this->getPrototypeChildren($node);
             }
 
-            if (!$children && !($node->hasDefaultValue() && \count($defaultArray = $node->getDefaultValue()))) {
-                $default = '[]';
+            if (!$children && !($node->hasDefaultValue() && \count(($defaultArray = $node->getDefaultValue()) ?? []))) {
+                $default = $node->hasDefaultValue() && $defaultArray === null ? '~' : '[]';
```

Comment:
This could be simplified by returning an empty array when the default is `null`, but I'm not sure it will meet expectations.
```suggestion
                $default = '[]';
```